### PR TITLE
fix: have set_routing_graph at interface level

### PR DESCRIPTION
**Pull request by JoanFM**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200479679796904/1200498812131103) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by hanxiao — CHANGES_REQUESTED** [review]
there is no "GatewayPod"

File: jina/peapods/pods/__init__.py [hunk 1]
```diff
@@ -265,6 +265,15 @@ def is_singleton(self) -> bool:
         """
         ...
 
+    def set_routing_graph(self, *args, **kwargs):
+        """
+        .. # noqa: DAR102
+        .. # noqa: DAR101
+        """
+        raise NotImplementedError(
+            'Set routing graph should only be called for `GateWayPod`'
```
Comment (JoanFM) [inline jina/peapods/pods/__init__.py:274]:
```suggestion
            'Set routing graph should only be called for `Pod`'
```

**Comment by hanxiao** [comment]
IDE complains of what?

**Comment by hanxiao** [comment]
simply `self._pod_nodes = OrderedDict()  # type: Dict[str, Pod]`

**Comment by JoanFM** [comment]
> IDE complains of what?

that BasePod does not have a set routing Graph method. Strictly speaking a CompoundPod instance could be part of the Flow and does not have it implemented.

**Comment by hanxiao** [comment]
this function has been removed completely in https://github.com/jina-ai/jina/pull/2726 and move to inline

**Pull request closed without merging**